lib/shared/puppet_forge/v3/module_release.rb

```diff
@@ -0,0 +1,72 @@
+require 'shared/puppet_forge/connection'
+
+module PuppetForge
+  module V3
+    # Access metadata and downloads for a specific module release.
+    class ModuleRelease
+
+      include PuppetForge::Connection
+
+      # @!attribute [r] full_name
+      #   @return [Forge::ModuleName] The hyphen delimited name of this module
+      attr_reader :full_name
+
+      # @!attribute [r] version
+      #   @return [String] The version of this module
+      attr_reader :version
+
+      # @param full_name [String] The name of the module, will be converted to
+      #   a PuppetForge::ModuleName instance.
+      # @param version [String]
+      def initialize(full_name, version)
+        @full_name = PuppetForge::ModuleName.new(full_name)
+        @version   = version
+      end
+
+      # @return [Hash] The metadata for this module
+      def data
+        @data ||= conn.get(resource_url).body
+      end
```

Comment:
Long term we should have a more structured way of accessing the response, but this is probably fine for now. The comment should probably not claim this is the module metadata though since that is an actual specific part of the response. :)